core/chains/evm/txm/orchestrator.go

```diff
@@ -0,0 +1,364 @@
+			return
+		}
+		o.txm.Trigger(request.FromAddress)
+	}
+
+	if wrappedTx.ID > math.MaxInt64 {
+		return tx, fmt.Errorf("overflow for int64: %d", wrappedTx.ID)
+	}
+
+	tx = txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee]{
+		ID:             int64(wrappedTx.ID),
+		IdempotencyKey: wrappedTx.IdempotencyKey,
+		FromAddress:    wrappedTx.FromAddress,
+		ToAddress:      wrappedTx.ToAddress,
+		EncodedPayload: wrappedTx.Data,
+		Value:          *wrappedTx.Value,
+		FeeLimit:       wrappedTx.SpecifiedGasLimit,
+		CreatedAt:      wrappedTx.CreatedAt,
+		Meta:           wrappedTx.Meta,
+		// Subject: wrappedTx.Subject,
+
+		// TransmitChecker: wrappedTx.TransmitChecker,
+		ChainID: wrappedTx.ChainID,
+
+		PipelineTaskRunID: wrappedTx.PipelineTaskRunID,
+		MinConfirmations:  wrappedTx.MinConfirmations,
+		SignalCallback:    wrappedTx.SignalCallback,
+		CallbackCompleted: wrappedTx.CallbackCompleted,
+	}
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) CountTransactionsByState(ctx context.Context, state txmgrtypes.TxState) (uint32, error) {
+	addresses, err := o.keystore.EnabledAddressesForChain(ctx, o.chainID)
+	if err != nil {
+		return 0, err
+	}
+	total := 0
+	for _, address := range addresses {
+		_, count, err := o.txStore.FetchUnconfirmedTransactionAtNonceWithCount(ctx, 0, address)
+		if err != nil {
+			return 0, err
+		}
+		total += count
+	}
+
+	//nolint:gosec // disable G115
+	return uint32(total), nil
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindEarliestUnconfirmedBroadcastTime(ctx context.Context) (time nullv4.Time, err error) {
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindEarliestUnconfirmedTxAttemptBlock(ctx context.Context) (time nullv4.Int, err error) {
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindTxesByMetaFieldAndStates(ctx context.Context, metaField string, metaValue string, states []txmgrtypes.TxState, chainID *big.Int) (txs []*txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee], err error) {
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindTxesWithMetaFieldByStates(ctx context.Context, metaField string, states []txmgrtypes.TxState, chainID *big.Int) (txs []*txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee], err error) {
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindTxesWithMetaFieldByReceiptBlockNum(ctx context.Context, metaField string, blockNum int64, chainID *big.Int) (txs []*txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee], err error) {
+	return
+}
+
+//nolint:revive // keep API backwards compatible
+func (o *Orchestrator[BLOCK_HASH, HEAD]) FindTxesWithAttemptsAndReceiptsByIdsAndState(ctx context.Context, ids []int64, states []txmgrtypes.TxState, chainID *big.Int) (txs []*txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee], err error) {
+	return
+}
```

Comment:
Is it ok to merge without having these implemented? Think these were added for VRF IIRC which I guess won't be using TXMv2 but wanted to double check.

Reply:
For now, yes, we don't have to worry about VRF support. As a side note, I view this as a future improvement: we shouldn't be treating the TXM as a transaction store by extending its interface when we want a new category of transaction tracking. This has resulted in a bad design, where the TXM's interface has multiple passthrough methods that basically query the DB. Instead, we should create a more generalized method that returns a bigger subset of the required transactions and the rest of the filtering should be done on the application logic. The application should be responsible for tracking most of the transactions and the TXM should be responsible for reporting the status, instead of implementing queries for metrics.